src/providers/ffz/FfzEmotes.cpp

```diff
@@ -47,68 +52,94 @@ namespace {
                                             : Image::fromUrl(url3x, 0.25)};
         emoteData.tooltip = {tooltip};
     }
+
     EmotePtr cachedOrMake(Emote &&emote, const EmoteId &id)
     {
         static std::unordered_map<EmoteId, std::weak_ptr<const Emote>> cache;
         static std::mutex mutex;
 
         return cachedOrMakeEmotePtr(std::move(emote), cache, mutex, id);
     }
-    std::pair<Outcome, EmoteMap> parseGlobalEmotes(
-        const QJsonObject &jsonRoot, const EmoteMap &currentEmotes)
+
+    QString kindToString(FfzEmoteKind kind)
+    {
+        switch (kind)
+        {
+            case FfzEmoteKind::Channel:
+                return QStringLiteral("Channel");
+            case FfzEmoteKind::Global:
+                return QStringLiteral("Global");
+            default:
+                return QStringLiteral("(invalid kind)");
+        }
+    }
+
+    void parseEmoteSetInto(const QJsonObject &emoteSet, FfzEmoteKind kind,
```

Comment:
Let's accept the emote kind as a string instead of an enum, since all we do with it is convert it to a string